LiskHQ/lisk-sdk
# Update FCS to check active delegate or not with 2 rounds setting

### What was the problem?

This PR resolves #5115 

### How was it solved?

- Revert the changes of checking 2 rounds for the FCS
- Update FCS trigger to check if the block received is from active delegate
- Remove the block slot and state check for the FCS mechanism while querying the next blocks
- Add isActiveDelegate function to DPoS using voteWeight
- Remove validateDetached, and consolidate with `validate` in the processor
- Cleanup unused function and combine them in verify step in the chain

### How was it tested?

- Start 2 nodes, and 1 node should have longer chain, then observe
  1. First block synchronization mechanism is triggered
  2. Fast chain switching is triggered
  3. stop the node for 1.5 blocks
  4. start again and check if the fast chain switching is triggered
